src/habakkuk/dag.py

```diff
@@ -60,9 +60,63 @@ def subgraph_matches(node1, node2):
     return matches
 
 
+def prune_array_index_constants(node):
+    ''' Prune sub-graphs representing "+/- constant" from the tree with
+    node at its root. Takes the root node of an array-index expression as
+    argument and returns the (potentially) new root node of the
+    expression. Since we are dealing with array-index expressions, any
+    array references we encounter are not modified. i.e. if we have the
+    expression "ji+1+map(ji+1)" then this routine will produce
+    "ji+map(ji+1)". '''
+
+    if node.node_type not in ["+", "-"]:
+        return node
```

Comment:
Needs a comment: "If the root node is not a + or - then we cannot prune any constants from the tree"